.github/workflows/ci.yml

```
@@ -0,0 +1,56 @@
+      - name: Run type checking
+        run: npx tsc --noEmit
+
+      - name: Run linting
+        run: npx eslint .
+
+      - name: Run tests with coverage
+        run: npx jest --coverage --ci --bail --runInBand
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v5
+        if: ${{ always() && (hashFiles('coverage/lcov.info') != '') && (secrets.CODECOV_TOKEN != '') }}
```

actionlint (1.7.7)

context "secrets" is not allowed here. available contexts are "env", "github", "inputs", "job", "matrix", "needs", "runner", "steps", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)